src/models.ts

```diff
@@ -54,15 +54,15 @@ export namespace Models {
     /**
      * Teams List
      */
-    export type TeamList = {
+    export type TeamList<Preferences extends Models.Preferences> = {
```

Comment:
Why does it have to be `Preferences extends Models.Preferences` and not just `Models.Preferences` ?